### PR TITLE
Fix MovingWindow localDimension

### LINKED ISSUE #568: MovingWindow localDimension Values Not Updated
The x and z component of `movingWindow.localDimension` is always zero.

BugFix MovingWindow.hpp:223
```C++
window.localDimensions = Selection<simDim>(subGrid.getLocalDomain().size);
```
to
```C++
window.localDimensions = Selection<simDim>(subGrid.getLocalDomain());
```

I have not checked any side effects with HDF5 or adios.

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- close #568 *MovingWindow localDimension Values Not Updated*
- fix that offsets for local and global dimensions always set to zero
- copy full selction from `SubGrid` instead only the size
- remove hard overwrite of the global window offset by add tu current value

Additional to the `window.localDimensions.offset` the `window.globalDimensions.offset` was also always zero.

- [x] merge after #1272

Tests:
- [x] HDF5 checkpoints LWFA with moving window; simulate 2k steps and restart from step 2k (simulate to step 3k)
- [x] ADIOS checkpoints LWFA with moving window; simulate 2k steps and restart from step 2k (simulate to step 3k)
- [x] LWFA with moving window - HDF5 plugin enabled; simulate 4k steps (checked with HDFView  -> picture; slice each dimension)
- [x] LWFA with moving window - ADIOS plugin enabled; simulate 4k steps

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
Thanks to @ax3l for help with adios.
The data looks fine.
ready for a review

File: src/picongpu/include/simulationControl/MovingWindow.hpp [hunk 1]
```diff
@@ -231,8 +231,8 @@ class MovingWindow
             float_64 offsetFirstGPU = 0.0;
             getCurrentSlideInfo(currentStep, NULL, &offsetFirstGPU);
 
-            /* global offset is all 0 except for y dimension */
-            window.globalDimensions.offset.y() = offsetFirstGPU;
+            /* moving window can only slide in y direction */
+            window.globalDimensions.offset.y() += offsetFirstGPU;
```
Comment (ax3l) [inline src/picongpu/include/simulationControl/MovingWindow.hpp]: should stay "=" (see "globalWindow.offset" [definition](https://github.com/ComputationalRadiationPhysics/picongpu/wiki/PIConGPU-domain-definitions))

File: src/picongpu/include/simulationControl/MovingWindow.hpp [hunk 2]
```diff
@@ -220,8 +220,8 @@ class MovingWindow
         const SubGrid<simDim>& subGrid = Environment<simDim>::get().SubGrid();
         Window window;
 
-        window.localDimensions = Selection<simDim>(subGrid.getLocalDomain().size);
-        window.globalDimensions = Selection<simDim>(subGrid.getGlobalDomain().size);
+        window.localDimensions = Selection<simDim>(subGrid.getLocalDomain());
```
Comment (ax3l) [inline src/picongpu/include/simulationControl/MovingWindow.hpp]: pls remove `Selection<simDim>(` wrapping.

File: src/picongpu/include/simulationControl/MovingWindow.hpp [hunk 3]
```diff
@@ -220,8 +220,8 @@ class MovingWindow
...
+        window.globalDimensions = Selection<simDim>(subGrid.getGlobalDomain());
```
Comment (ax3l) [inline src/picongpu/include/simulationControl/MovingWindow.hpp]: keep, `.size`; offset of *global* window is equal to `offsetFirstGPU` an all GPUs

File: src/picongpu/include/simulationControl/MovingWindow.hpp [hunk 4]
```diff
@@ -220,8 +220,8 @@ class MovingWindow
         const SubGrid<simDim>& subGrid = Environment<simDim>::get().SubGrid();
         Window window;
```
Comment (ax3l) [inline src/picongpu/include/simulationControl/MovingWindow.hpp]: place a comment above:
```c++
/* Without moving window, the selected window spans the whole global domain.
 * \see https://github.com/ComputationalRadiationPhysics/picongpu/wiki/PIConGPU-domain-definitions
 *
 * The window's global offset is therefore zero inside the global domain. The window's global
 * and local size are equal to the SubGrid quantities. The local window offset is the offset within
 * the global window which is equal to the local domain offset of the GPU. */
```

File: src/picongpu/include/simulationControl/MovingWindow.hpp [hunk 5]
```diff
@@ -220,8 +220,8 @@ class MovingWindow
...
 
-        window.localDimensions = Selection<simDim>(subGrid.getLocalDomain().size);
-        window.globalDimensions = Selection<simDim>(subGrid.getGlobalDomain().size);
+        window.localDimensions = Selection<simDim>(subGrid.getLocalDomain());
+        window.globalDimensions = Selection<simDim>(subGrid.getGlobalDomain());
 
         /* If sliding is inactive, moving window is the same as global domain (substract 0)*/
         window.globalDimensions.size.y() -= subGrid.getLocalDomain().size.y() * slidingWindowActive;
```
Comment (ax3l) [inline src/picongpu/include/simulationControl/MovingWindow.hpp]: offline: move down to `if`

File: src/picongpu/include/simulationControl/MovingWindow.hpp [hunk 6]
```diff
@@ -220,8 +220,8 @@ class MovingWindow
         const SubGrid<simDim>& subGrid = Environment<simDim>::get().SubGrid();
         Window window;
 
-        window.localDimensions = Selection<simDim>(subGrid.getLocalDomain().size);
-        window.globalDimensions = Selection<simDim>(subGrid.getGlobalDomain().size);
+        window.localDimensions = Selection<simDim>(subGrid.getLocalDomain());
+        window.globalDimensions = Selection<simDim>(subGrid.getGlobalDomain());
 
         /* If sliding is inactive, moving window is the same as global domain (substract 0)*/
```
Comment (ax3l) [inline src/picongpu/include/simulationControl/MovingWindow.hpp]: new comment: `/* the moving window is smaller than the global domain by *one GPU* in moving (y) direction */`

File: src/picongpu/include/simulationControl/MovingWindow.hpp [hunk 7]
```diff
@@ -231,8 +231,8 @@ class MovingWindow
             float_64 offsetFirstGPU = 0.0;
             getCurrentSlideInfo(currentStep, NULL, &offsetFirstGPU);
 
-            /* global offset is all 0 except for y dimension */
-            window.globalDimensions.offset.y() = offsetFirstGPU;
+            /* moving window can only slide in y direction */
```
Comment (ax3l) [inline src/picongpu/include/simulationControl/MovingWindow.hpp]: move this comment before `if (slidingWindowActive)`

File: src/picongpu/include/simulationControl/MovingWindow.hpp (hunk 1 above)
Comment (ax3l) [inline src/picongpu/include/simulationControl/MovingWindow.hpp]: new comment:
```c++
/* while moving, the windows global offset within the global domain is between 0
 * and smaller than the local domain's size in y.
 */
```

**Comment by ax3l** [comment]
let us remove line 246 `window.localDimensions.offset.y() = 0;` since it *is* already zero for the top GPU

**Comment by ax3l** [comment]
line 251 to:
```c++
/* the windows local offset within the global window is reduced by the global window offset
 * within the global domain
 */
window.localDimensions.offset.y() -= offsetFirstGPU;
```

**Comment by psychocoderHPC** [comment]
ready to merge

**Pull request merged**